### PR TITLE
Sheet selector

### PR DESCRIPTION
**Pull request by Erik9519**
#### Deployment instructions
None.

#### What have I done
Fixed Import tests.
Implemented Sheet Selector.
Changes to Sheet Service to better suit the Sheet Selector.

#### Testing
JUnit testing only.

### CODE REVIEW

File: src/app/backend/sheet-service/sheet.service.ts [hunk 1]
```diff
@@ -9,18 +9,30 @@ export class SheetService {
...
+     */
+    @Output() newSelectedSheet = new EventEmitter<Sheet>();
 
     /**
      * Set the available sheets
      * @param sheets list of sheets
      */
     setSheets(sheets: Sheet[]) {
         this.sheets = sheets;
+        this.selectedSheet = sheets[0];
```
Comment (Dacaspex) [inline src/app/backend/sheet-service/sheet.service.ts]: I would not select a sheet when new sheets are available

File: src/app/toolbar/import/import.component.spec.ts [hunk 2]
```diff
@@ -50,28 +46,31 @@ describe('ImportComponent', () => {
...
-  it('load timing file', () => {
+  it('load timing file', (done) => {
       const file = new Blob(['hello']);
       // Simulate input[type=file} event contents
       const event = { target: {files: [file]} };
       // Fire Event
       component.onFileChange(event, 'timing');
       // Wait for the file to be loaded (500ms max)
       setTimeout(() => {
+          fixture.detectChanges();
```
Comment (Dacaspex) [inline src/app/toolbar/import/import.component.spec.ts]: Should this be gone? It is not at the other test cases

File: src/app/toolbar/sheet-selector/sheet-selector.component.ts [hunk 3]
```diff
@@ -1,19 +1,72 @@
 import { Component, OnInit } from '@angular/core';
+import { SheetService } from '../../backend/sheet-service/sheet.service';
+import { Sheet } from '../../backend/models/sheet/sheet';
 
 @Component({
     selector: 'app-sheet-selector',
     templateUrl: './sheet-selector.component.html',
     styleUrls: ['./sheet-selector.component.scss']
 })
-export class SheetSelectorComponent implements OnInit {
```
Comment (Dacaspex) [inline src/app/toolbar/sheet-selector/sheet-selector.component.ts]: Newline too much

File: src/app/toolbar/sheet-selector/sheet-selector.component.ts [hunk 4]
```diff
@@ -1,19 +1,72 @@
...
 
-    sheets = ['sheet 1', 'sheet 2', 'sheet 3']; // TODO dynamically load sheets.
-    selectedSheet;
+export class SheetSelectorComponent implements OnInit {
+    /**
+     * Contains the sheet names to display in the drop down
+     */
+    sheets = [];
```
Comment (Dacaspex) [inline src/app/toolbar/sheet-selector/sheet-selector.component.ts]: Type `string[]`

File: src/app/toolbar/sheet-selector/sheet-selector.component.ts [hunk 5]
```diff
@@ -1,19 +1,72 @@
...
     ngOnInit() {
+        // Not Required
     }
 
+    /**
+     * This event fires whenever new sheets are available
+     */
+    updateSheetSelector() {
+        // Clear Array
+        this.sheets.length = 0;
```
Comment (Dacaspex) [inline src/app/toolbar/sheet-selector/sheet-selector.component.ts]: Why not use `this.sheets = [];`?

File: src/app/toolbar/sheet-selector/sheet-selector.component.ts [hunk 6]
```diff
@@ -1,19 +1,72 @@
...
+        this.sheetIds.length = 0;
+        // Populate Array
+        if (this.sheetService.getSheets() !== undefined) {
+            this.sheetService.getSheets().forEach((sheet: Sheet) => {
+                this.sheetIds.push(sheet.getId());
+                this.sheets.push('Sheet ' + sheet.getId());
+            });
+            // Automatically select sheet
+            const selectedSheetIndex = this.sheetIds.findIndex((id: number) => {
```
Comment (Dacaspex) [inline src/app/toolbar/sheet-selector/sheet-selector.component.ts]: There should not be a pre-selected sheet at startup (or when a new model is loaded)

File: src/app/toolbar/sheet-selector/sheet-selector.component.ts [hunk 7]
```diff
@@ -1,19 +1,72 @@
...
+                return (id === this.sheetService.getSelectedSheet().getId());
+            });
+            this.selectedSheet = this.sheets[selectedSheetIndex];
+        } else {
+            this.sheets.length = 0;
```
Comment (Dacaspex) [inline src/app/toolbar/sheet-selector/sheet-selector.component.ts]: Same thing here with `[]`

File: src/app/backend/sheet-service/sheet.service.ts (hunk 1 above)
Comment (Erik9519) [reply src/app/backend/sheet-service/sheet.service.ts]: Changed to select nothing

**Pull request merged**